app/components/pages/SubmissionWizard/steps/Editors/SuggestedReviewersValidator.js

```diff
@@ -0,0 +1,132 @@
+import yup, { ValidationError } from 'yup'
+import { cloneDeep, countBy, findIndex } from 'lodash'
+
+const MIN_REVIEWERS = 3
+const MAX_REVIEWERS = 6
+
+export const suggestedReviewersLimits = {
+  min: MIN_REVIEWERS,
+  max: MAX_REVIEWERS,
+}
+
+class SuggestedReviewersValidator {
+  constructor() {
+    this.schema = yup.object({
+      name: yup.string().required('Name is required'),
+      email: yup
+        .string()
+        .email('Must be a valid email')
+        .required('Email is required'),
+    })
+    this.thingsNotDuplicated = ['name', 'email']
+  }
+
+  checkArrayContents(reviewers, errors) {
```

Comment:
I'm wondering if there's a way to pop all of this `SuggestedReviewsValidator` logic into the component itself? So that there's yup can be responsible only for doing the checking, rather than whether something should be checked?

Reply:
So  the latest PR has all the add/removal done in the UI ... and all the validation in this class.